File: solutions/python/luhn/1/luhn.py
class Luhn:
    def __init__(self, card_num):
        self.card_num = card_num.replace(" ", "")

    def valid(self):
        l = len(self.card_num)
        if l < 2:
            return False
        result = 0
        for i in range(0, l):
            char = self.card_num[-1 - i]
            if char.isnumeric():
                number = int(char)
                if i % 2 != 0:
                    number <<= 1
                    number = number - 9 if number > 9 else number
                result += number
            else:
                return False
        return result % 10 == 0

File: solutions/javascript/luhn/1/luhn.js
//
// This is only a SKELETON file for the 'Luhn' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const valid = (n) => {
  n = n.replace(/\s/g, '');
  const len = n.length;
  if(len <= 1 || !/^\d+$/.test(n))
    return false;
  let sum = 0, flag = false;
  for(let i = len - 1; i >= 0; i--){
    if(flag){
      const double = Number(n[i]) * 2;
      sum += double > 9 ? double - 9 : double;
    } else 
      sum += Number(n[i]);
    flag = !flag;
  }
  return sum % 10 === 0;
};
